src/client/components/copybutton/copybutton.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {Button} from 'reactstrap';
+import { copyToClipboard } from '../../../utils/utils';
+import PropTypes from 'prop-types';
+
+class CopyButton extends React.Component {
+    static propTypes = {
+        color: PropTypes.string.isRequired
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {color: this.props.color};
+    }
+
+    render() {
+        return (
+            <Button color={this.state.color} onClick={() => this.handleClick()}>
+                Copy
+            </Button>
+        )
+    }
+
+    handleClick() {
+        copyToClipboard(this.state.text).then(() => {
```

Comment:
`this.state.text` is not defined. Therefore, not the link but the string `undefined` gets copied.

Reply:
Whoops, I meant text to be a property